### PR TITLE
3scale-2.10

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
## Verification Steps
As the verifier of the PR the following process should be done:
1. Install from master
2. upgrade from this PR

### Installation Verification
- Ensure the author of the PR has attached a log of the installation run from his branch to the jira or pr and check that it exited as expected.
- Verify the fresh installation is correct on cluster provided by PR author 
### Upgrade Verification
- After installation verification, notify the PR author to begin an upgrade on their cluster
- Ensure the developer of the PR has attached a log of the upgrade run from his branch to the jira or pr and check that it exited as expected. 
- If possible, look at the tasks that ran and see they match the PR
- Verify the upgrade is correct on cluster provided by PR author 

## Checklist

- [ ] Updated [Changelog](https://github.com/integr8ly/installation/blob/master/CHANGELOG.md)
- [ ] Tested Installation
- [ ] Tested Uninstallation
- [ ] Tested or Created follow on for Upgrade

Is an upgrade task required and are there additional steps needed to test this?
- [ ] Yes
- [ ] No

### CODE REVIEW

File: inventories/group_vars/all/manifest.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ integreatly_version: master
 threescale: True
 # below threescale variables below are not currently used. They define where and what to pull in from github. These will be important as we look to break out and decouple the resources from the installer ###############
 # not currently used but records the version installed by the operator, this should come from the status block
-threescale_version: '2.9.0.GA'
+threescale_version: '2.10.0.GA'
```
Comment (jessesarn) [inline inventories/group_vars/all/manifest.yaml:10]:
```suggestion
threescale_version: '3scale-2.10.0-GA'
```
Apparently 3scale changed their tagging convention to include a 3scale- prefix starting with 2.10.0.  without this, install of 2.10 on fresh cluster won't work.

File: roles/3scale/defaults/main.yml [hunk 2]
```diff
@@ -154,14 +154,14 @@ threescale_route_system_master: system-master
 threescale_route_system_provider: system-provider
 
 threescale_image_streams:
-  - amp-apicast:latest
```
Comment (jessesarn) [inline roles/3scale/defaults/main.yml:157]: These lines should remain latest.  The threescale_image_streams var is only used during upgrade_images to pull down the new versions.  So from an upgrade of 3scale 2.9 on RHMI 1.9 to 3scale 2.10 on RHMI 1.10 would want to use latest to get 2.10.

**Review by jessesarn — CHANGES_REQUESTED** [review]

**Comment by jessesarn** [comment]
Something seem to be broken with the use of latest with the new 2.10 release.  I installed release-1.9.0 (with the fixes to login and avoid the sso cert issues on rhpds), set upgrade_from_version=release-1.9.0 and then tried to run upgrade with this PR and got:

```
<127.0.0.1> EXEC /bin/sh -c 'echo ~root && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1619205308.35-124783112414303 `" && echo ansible-tmp-1619205308.35-124783112414303="` echo /root/.ansible/tmp/ansible-tmp-1619205308.35-124783112414303 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/commands/command.py
<127.0.0.1> PUT /root/.ansible/tmp/ansible-local-8997AyD5eE/tmpOpnGyr TO /root/.ansible/tmp/ansible-tmp-1619205308.35-124783112414303/command.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1619205308.35-124783112414303/ /root/.ansible/tmp/ansible-tmp-1619205308.35-124783112414303/command.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1619205308.35-124783112414303/command.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1619205308.35-124783112414303/ > /dev/null 2>&1 && sleep 0'
failed: [127.0.0.1] (item=backend-redis:latest) => {
    "changed": true, 
    "cmd": "oc import-image backend-redis:latest -n 3scale", 
    "delta": "0:00:00.178697", 
    "end": "2021-04-23 19:15:08.652845", 
    "invocation": {
        "module_args": {
            "_raw_params": "oc import-image backend-redis:latest -n 3scale", 
            "_uses_shell": true, 
            "argv": null, 
            "chdir": null, 
            "creates": null, 
            "executable": null, 
            "removes": null, 
            "stdin": null, 
            "warn": true
        }
    }, 
    "msg": "non-zero return code", 
    "patch_image_stream_item": "backend-redis:latest", 
    "rc": 1, 
    "start": "2021-04-23 19:15:08.474148", 
    "stderr": "error: the tag \"latest\" does not exist on the image stream - choose an existing tag to import or use the 'tag' command to create a new tag", 
    "stderr_lines": [
        "error: the tag \"latest\" does not exist on the image stream - choose an existing tag to import or use the 'tag' command to create a new tag"
    ], 
    "stdout": "", 
    "stdout_lines": []
}

PLAY RECAP ******************************************************************************************************************************************************************************************************************************************************************************************************************
127.0.0.1                  : ok=9    changed=3    unreachable=0    failed=1   
```
So I think we broken upgrade when we fixed install or something along those lines.

Full upgrade log: https://gist.github.com/jessesarn/c06f5b2c8d0584d4b7a977061a9a83d3

**Comment by CathalOConnorRH** [comment]
tested upgrade with `latest` in roles/3scale/defaults/main.yml and it failed, tested with 2.10 and it failed, tested with 2.9, ie the current image on cluster and it worked. upgrade log here https://gist.github.com/CathalOConnorRH/dea5f52540cd804c0cd7bcc2b5bb3544

**Comment by jessesarn** [comment]
Tested install of 1.9.0 and then upgrade to 3scale 2.10 with this PR.
Tested full install from this PR.

All 3 logs can be found here: https://gist.github.com/jessesarn/15d88983dd2b63b973fcad1001751ae7

**Review by jessesarn — APPROVED** [review]

**Pull request merged**